components/org.wso2.carbon.identity.media.endpoint/src/main/java/org/wso2/carbon/identity/media/endpoint/service/MediaService.java

```diff
@@ -0,0 +1,242 @@
+        String tenantDomain = getTenantDomainFromContext();
+        try {
+            String uuid = storageSystemManager.addFile(filesInputStream, mediaMetadata, tenantDomain);
+            if (StringUtils.isNotBlank(uuid)) {
+                return uuid;
+            }
+            throw handleException(Response.Status.INTERNAL_SERVER_ERROR,
+                    MediaServiceConstants.ErrorMessage.ERROR_CODE_ERROR_UPLOADING_MEDIA);
+        } catch (StorageSystemException e) {
+            MediaServiceConstants.ErrorMessage errorMessage = MediaServiceConstants.ErrorMessage.
+                    ERROR_CODE_ERROR_UPLOADING_MEDIA;
+            Response.Status status = Response.Status.INTERNAL_SERVER_ERROR;
+            throw handleException(e, errorMessage, LOG, status);
+        }
+    }
+
+    private void setSecurityForEndUserUploadedMedia(Metadata metadata, MediaMetadata mediaMetadata) {
+
+        Security securityMeta = null;
+        if (metadata != null) {
+            securityMeta = metadata.getSecurity();
+        }
+        FileSecurity fileSecurity;
+        if (securityMeta != null && securityMeta.getAllowedAll()) {
+            fileSecurity = new FileSecurity(true);
+        } else {
+            ArrayList<String> users = new ArrayList<>();
+            users.add(getUsernameFromContext());
+            fileSecurity = new FileSecurity(false, users);
+        }
+        mediaMetadata.setFileSecurity(fileSecurity);
+    }
+
+    private void setSecurityForPrivilegedUserUploadedMedia(PrivilegedUserMetadata privilegedUserMetadata,
+                                                           MediaMetadata mediaMetadata) {
+
+        PrivilegedUserSecurity fileSecurityMeta = null;
+        if (privilegedUserMetadata != null) {
+            fileSecurityMeta = privilegedUserMetadata.getSecurity();
+        }
+        FileSecurity fileSecurity;
+        if (fileSecurityMeta != null && fileSecurityMeta.getAllowedAll()) {
+            fileSecurity = new FileSecurity(true);
+        } else {
+            List<String> allowedUsers = null;
+            if (fileSecurityMeta != null) {
+                allowedUsers = fileSecurityMeta.getAllowedUsers();
+            }
+            if (CollectionUtils.isNotEmpty(allowedUsers)) {
+                fileSecurity = new FileSecurity(false, allowedUsers);
+            } else {
+                fileSecurity = new FileSecurity(false);
+            }
+        }
+        mediaMetadata.setFileSecurity(fileSecurity);
+    }
+
+    /**
+     * Validate if the uploaded media content type is a supported media type.
+     *
+     * @param mediaTypePathParam The media type available as a path parameter in the upload request.
+     * @param mediaType          The content type of the uploaded media.
+     */
+    public void validateMediaType(String mediaTypePathParam, MediaType mediaType) {
+
+        if (!mediaTypePathParam.equals(mediaType.getType())) {
```

Comment:
Better to use StringUtils here. Otherwise, there is a possibility of NPE.

Reply:
fixed with: f3c2d735de5ca72f35a49a5e293a33c14179262c